### DISCUSSION TITLE
Status "Needs-Action" and progress value

### DISCUSSION THREAD

**Discussion by transeuronight** [TechbeeAT/jtxBoard#612] [Ideas]
Since a couple of weeks I'm using jtxBoard together with DavX5 on Android and I consider the task app as the best what I see so far on Android.

However, there is one synchronization problem concerning tasks with the Status _Needs-Action_ according [RFC5545 Section 3.8.1.11](https://www.rfc-editor.org/rfc/rfc5545#section-3.8.1.11). 
If I choose this status in jtxBoard the progress value returns to 0, which is
a) not the case in Thunderbird/Lightning
b) not specified, neither in RFC5545 nor in the latest draft for [Task Extensions to iCalendar](https://datatracker.ietf.org/doc/draft-ietf-calext-ical-tasks/)

Maybe there is a misunderstanding, since the german translation of the NEEDS-ACTION status is "Aktion ausständig" which could be also interpreted as the need for an action **before** the task can be started.
Indeed the english term "NEEDS-ACTION" is agnostic concerning the progress, since the need for an action can be at the beginning as well as during a task - therefore I see no reason to reset the progress value to 0.

Therefore I hope that this behaviour changes, so that the status of NEEDS_ACTION doesn't affect the progress value. If someone still wants to reset it to 0, he/she can do it manually but currently there is no possibility to keep the progress value with this status.

**Comment by rogercreagh** [comment]
Ah-ha - this relates to a previous discussion about status (which I can't now find -I think it was about Kanban columns) where I said I thought NEEDS-ACTION does not equate to progress 0 and @patrickunterwegs said he thought it did. 

Now all is explained perhaps - as @transeuronight says the english phrase NEEDS-ACTION is agnostic about progress. But perhaps the german version doesn't accurately reflect the meaning of the english. 

For an Englsh speaker NEEDS-ACTION could be applied at any state of progress less than 100% complete - and typically might imply that there is some external action (outside the project) which has to take place before further progress can be made (eg appointment of a specialist, or waiting for management sign-off).

So I 100% agree with @transeuronight that changing status to NEEDS-ACTION should NOT reset progress to 0. If it comes down to language wars then I think the master language for the RFC is English? :-)

**Comment by patrickunterwegs** [comment]
Yes, alright, I agree with you, I might have over-interpreted the standard in this case. And also I'd adapt the German translation to "Aktion erforderlich"

**Reply by patrickunterwegs** [reply]
And thanks for your feedback :)

**Reply by transeuronight** [reply]
Thanks for the adoption!

**Comment by patrickunterwegs** [comment]
@rogercreagh , @transeuronight , I have almost finished the change, but what would you say if the status is set back to "no status"? Would you expect the progress to be reset to 0? Or stay unchanged?

**Reply by bepolymathe** [reply]
Hi, 

If I can join in the discussion... Thunderbird behaves like this. If you go back to "unspecified" status, progress is reset to zero. You can no longer change the progress. This is consistent with Thunderbird. Evolution has the same behavior, but you can assign a progress percentage to a task in "unspecified" status. Personally, I find it handy to be able to assign a percentage to any status independently. Evolution seems to me to have the "best" behavior from this point of view.

**Comment by rogercreagh** [comment]
Personally I think Status should be independent from Progress.

If there is to be a link it should be an optional user-defined progress value for each status.

By default status values would  have a progress value of null -meaning no change to progress when the status is applied.
The user (possibly only with admin privileges if there is a security model in play) should be able to define a progress percent value for any status value. 
The progress would be set to this value when the status value for a task is changed if the new status has an associated progress percent
(?) Defining or changing the progress value associated with a status would not change any existing tasks progress that already had that status
The status progress values would be global across all collections (or it would get horribly complex to manage)

Any imposed, not user defined, linking of a status (or an unset status) toa specific value (or zero) is going to  be a serious annoyance for some users and useful for others. So it seems to me the only way to proceed is either to have no linkage, or to have a user defined linkage.

There is also the question of what you should do if a new task created on another app has no status but does have a progress >0% - it would make no sense to change the progress automatically back to zero in jtxB when it had been set to a specific value elsewhere.

If progress is gong to be linked to (some) status values then the user is still able to change the progress away from the preset value. It should only change automatically if a new status is applied.

**Reply by rogercreagh** [reply]
In short, I would find the Thunderbird handling of no-status very annoying. I should always be able to change the progress value whatever the status is set to.

Since there is no link between status and progress in the spec I would find any automatic change of progress when I changed the status very annoying, UNLESS I had specified the progress value that a particular status should represent.

The only possible exception might be setting the progress to 100% could set the status to completed (and vice versa)

**Reply by transeuronight** [reply]
Unfortunately "No-Status" is the only status which is completely undefineded.

All other status have actually a well-defined behaviour.

- NEEDS-ACTION and IN-PROCESS remain the status value untouched
- CANCELLED deactivates the status area but remains the status untouchted, too
- COMPLETED set the status value to 100% which makes sense

Since No-Status is at least not defined according [RFC5545 Section 3.8.1.11](https://www.rfc-editor.org/rfc/rfc5545#section-3.8.1.11) you can argue with the meaning itself "No Status" means no status information at all - therefore status value 0 (like Thunderbird). But I can also follow the argumentation of @rogercreagh - especially if another program interprets this status in a different manner. 
Therefore I'm not in favor of any solution and can't help - sorry @patrickunterwegs

**Reply by patrickunterwegs** [reply]
Thanks @transeuronight @rogercreagh . 
I will do the following: 
The progress would only update if you change the status to COMPLETED (100%) or IN-PROCESS (at least 1 %). And also the other way round, the status would only update when the progress is set to 1-99 % (IN-PROCESS) or 100 % (COMPLETED).

There is still the option to deactivate this behavior in the settings menu to keep both fields completely independent.
